### ISSUE TITLE
Detect UnicodeDecodeErrors when parsing config file

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#609]
In GitLab by @sigmavirus24 on Aug 1, 2017, 04:41

*Please describe how you installed Flake8*

Not applicable

*Please provide the exact, unmodified output of `flake8 --bug-report`*

Not applicable

*Please describe the problem or feature*

See also https://stackoverflow.com/questions/45401877/fix-error-unicodedecodeerror-with-flake8

If a config file has characters that we cannot handle, we should catch the UnicodeDecodeError and raise a new error to help the user understand what went wrong.

*If this is a bug report, please explain with examples (and example code) what you expected to happen and what actually happened.*

**Comment by asottile** [comment]
In GitLab by @asottile on Oct 9, 2017, 08:26

Since the SO post has been deleted (and can only be seen by 10k+ users), here's the trace:


```python
    Traceback (most recent call last):
      File "c:\python36\lib\runpy.py", line 193, in _run_module_as_main
        "__main__", mod_spec)
      File "c:\python36\lib\runpy.py", line 85, in _run_code
        exec(code, run_globals)
      File "C:\Python36\Scripts\flake8.exe\__main__.py", line 9, in <module>
      File "c:\python36\lib\site-packages\flake8\main\cli.py", line 16, in main
        app.run(argv)
      File "c:\python36\lib\site-packages\flake8\main\application.py", line 348, in run
        self._run(argv)
      File "c:\python36\lib\site-packages\flake8\main\application.py", line 335, in _run
        self.initialize(argv)
      File "c:\python36\lib\site-packages\flake8\main\application.py", line 319, in initialize
        self.parse_configuration_and_cli(argv)
      File "c:\python36\lib\site-packages\flake8\main\application.py", line 168, in parse_configuration_and_cli
        self.option_manager, argv
      File "c:\python36\lib\site-packages\flake8\options\aggregator.py", line 47, in aggregate_options
        original_values.isolated)
      File "c:\python36\lib\site-packages\flake8\options\config.py", line 282, in parse
        return self.merge_user_and_local_config()
      File "c:\python36\lib\site-packages\flake8\options\config.py", line 246, in merge_user_and_local_config
        config = self.parse_local_config()
      File "c:\python36\lib\site-packages\flake8\options\config.py", line 206, in parse_local_config
        config = self.config_finder.local_configs()
      File "c:\python36\lib\site-packages\flake8\options\config.py", line 107, in local_configs
        config, found_files = self._read_config(self.local_config_files())
      File "c:\python36\lib\site-packages\flake8\options\config.py", line 56, in _read_config
        found_files = config.read(files)
      File "c:\python36\lib\configparser.py", line 697, in read
        self._read(fp, filename)
      File "c:\python36\lib\configparser.py", line 1015, in _read
        for lineno, line in enumerate(fp, start=1):
      File "c:\python36\lib\encodings\cp1251.py", line 23, in decode
        return codecs.charmap_decode(input,self.errors,decoding_table)[0]
    UnicodeDecodeError: 'charmap' codec can't decode byte 0x98 in position 119: character maps to <undefined>
```

**Comment by asottile** [comment]
In GitLab by @asottile on Oct 9, 2017, 09:28

mentioned in merge request !208

**Comment by asottile** [comment]
In GitLab by @sigmavirus24 on Oct 19, 2017, 05:27

closed
